addons/website_sale_wallet/models/partner.py

```diff
@@ -0,0 +1,80 @@
+# -*- coding: utf-8 -*-
+
+from odoo import fields, models, api
+import functools
+
+
+class Partner(models.Model):
+    _inherit = "res.partner"
+
+    # Relational fields
+    gift_card_ids = fields.One2many('gift.card', "partner_id")
+    order_line_gift_card_ids = fields.One2many("sale.order.line", "wallet_id")
+    currency_id = fields.Many2one('res.currency',
```

Comment:
currency is depending of pricelist applied and not from partner

Store the amount in the currency of the company and use it in the currency of pricelist (like for product price e.g.)